central/complianceoperator/v2/scanconfigurations/service/convert.go

```diff
@@ -206,11 +229,29 @@ func getLatestBindingError(status *storage.ComplianceOperatorStatus) string {
 
 func convertStorageScanConfigToV2ScanStatus(ctx context.Context,
 	scanConfig *storage.ComplianceOperatorScanConfigurationV2, configDS complianceDS.DataStore,
-	bindingsDS bindingsDS.DataStore, suiteDS suiteDS.DataStore) (*v2.ComplianceScanConfigurationStatus, error) {
+	bindingsDS bindingsDS.DataStore, suiteDS suiteDS.DataStore, notifierDS notifierDS.DataStore) (*v2.ComplianceScanConfigurationStatus, error) {
 	if scanConfig == nil {
 		return nil, nil
 	}
 
+	notifiers := []*v2.NotifierConfiguration{}
+	for _, notifierConfig := range scanConfig.GetNotifiers() {
+
+		notifier, found, err := notifierDS.GetNotifier(ctx, notifierConfig.GetId())
+		if err != nil {
+			return nil, err
+		}
+		if !found {
+			return nil, errors.Errorf("Notifier with ID %s no longer exists", notifierConfig.GetId())
+		}
+		notifierV2, err := convertProtoNotifierConfigToV2(notifierConfig)
```

Comment:
You are going to have to do one of two things here:

Either check `notifierV2` for nil before trying to set the name of it OR have the convert function return an error when there is not a notifier or not an email notifier.  

A notifier that is not an email notifier is probably an error.  
No notifier is a valid state.  

so before doing `notifierV2.NotifierName` you have to check for nil.

In fact it may even be better to pass the name to the `convertProtoNotifierConfigToV2` and simply set it when you create the object.